packages/bridge-ui/src/storage/customTokenService.ts

```diff
@@ -0,0 +1,47 @@
+import type { Token, TokenStore } from "src/domain/token";
+
+interface storage {
+  getItem(key: string): string;
+  setItem(key: string, value: string);
+}
+
+class CustomTokenService implements TokenStore {
+  private readonly storage: storage;
+
+  constructor(
+    storage: storage,
+  ) {
+    this.storage = storage;
+  }
+
+  async StoreToken(token: Token, address: string): Promise<Token[]> {
+    const customTokens = this.storage.getItem(`custom-tokens-${address.toLowerCase()}`);
+    let tokens = [];
+    if(customTokens) {
+      tokens = [...JSON.parse(customTokens)];
+    }
+    const doesTokenAlreadyExist = tokens.findIndex(t => t.symbol === token.symbol);
+    if(doesTokenAlreadyExist < 0) {
+      tokens.push({ ...token });
+    }
+    this.storage.setItem(`custom-tokens-${address.toLowerCase()}`, JSON.stringify(tokens));
+    return tokens;
+  }
+
+  GetTokens(address: string) {
```

Comment:
this method has an inferred return type but the StoreToken has a defined return type, can we make them consistent for domain methods? I usually prefer the defined return type but its up to you!